src/System.Private.CoreLib/shared/System/Globalization/CalendarData.Unix.cs

```diff
@@ -35,22 +36,25 @@ internal partial class CalendarData
         private bool LoadCalendarDataFromSystem(string localeName, CalendarId calendarId)
         {
             bool result = true;
-            result &= GetCalendarInfo(localeName, calendarId, CalendarDataType.NativeName, out this.sNativeName);
-            result &= GetCalendarInfo(localeName, calendarId, CalendarDataType.MonthDay, out this.sMonthDay);
+            // TODO-NULLABLE: these can return null but are later replaced with String.Empty or other non-nullable value
+            result &= GetCalendarInfo(localeName, calendarId, CalendarDataType.NativeName, out this.sNativeName!);
+            result &= GetCalendarInfo(localeName, calendarId, CalendarDataType.MonthDay, out this.sMonthDay!);
             this.sMonthDay = NormalizeDatePattern(this.sMonthDay);
 
-            result &= EnumDatePatterns(localeName, calendarId, CalendarDataType.ShortDates, out this.saShortDates);
-            result &= EnumDatePatterns(localeName, calendarId, CalendarDataType.LongDates, out this.saLongDates);
-            result &= EnumDatePatterns(localeName, calendarId, CalendarDataType.YearMonths, out this.saYearMonths);
-            result &= EnumCalendarInfo(localeName, calendarId, CalendarDataType.DayNames, out this.saDayNames);
-            result &= EnumCalendarInfo(localeName, calendarId, CalendarDataType.AbbrevDayNames, out this.saAbbrevDayNames);
-            result &= EnumCalendarInfo(localeName, calendarId, CalendarDataType.SuperShortDayNames, out this.saSuperShortDayNames);
+            result &= EnumDatePatterns(localeName, calendarId, CalendarDataType.ShortDates, out this.saShortDates!);
```

Comment:
I'm confused about these `EnumDatePatterns` calls, because the method actually sets datePatterns to null and after its call `EnumCalendarInfo` if result is false, it will stay as a `null` value.

Reply:
It won't:
https://github.com/dotnet/coreclr/blob/4e3b01417d2b6dcdac66a987335a14b922fa6787/src/System.Private.CoreLib/shared/System/Globalization/CalendarData.cs#L113